# Dia 7 — Revisão e GitHub
Consolidando 6 dias de aprendizado: Pandas, Visualização, Limpeza, ML, Classificação, Streamlit

In [3]:
import os
import json
from pathlib import Path

# Diretório do projeto
projeto_path = Path("../..")

# 1. Contar arquivos por tipo
def contar_arquivos(extensao):
    count = 0
    for root, dirs, files in os.walk(projeto_path):
        # Ignora .venv e .git
        dirs[:] = [d for d in dirs if d not in ['.venv', '.git', '__pycache__']]
        for file in files:
            if file.endswith(extensao):
                count += 1
    return count

notebooks = contar_arquivos('.ipynb')
scripts_python = contar_arquivos('.py')
markdown_docs = contar_arquivos('.md')
csv_files = contar_arquivos('.csv')

print("=== Estatísticas do Projeto ===\n")
print(f"Notebooks Jupter:    {notebooks}")
print(f"Scripts Python:      {scripts_python}")
print(f"Documentação (MD):   {markdown_docs}")
print(f"Datasets (CSV):      {csv_files}")

# 2. Tamanho total do projeto (excluindo .venv)
total_size = 0
for root, dirs, files in os.walk(projeto_path):
    dirs[:] = [d for d in dirs if d not in ['.venv', '.git']]
    for file in files:
        filepath = os.path.join(root, file)
        total_size += os.path.getsize(filepath)

size_mb = total_size / (1024 * 1024)        
print(f"\nTamanho total:     {size_mb:.1f} MB")

=== Estatísticas do Projeto ===

Notebooks Jupter:    7
Scripts Python:      8
Documentação (MD):   21
Datasets (CSV):      9

Tamanho total:     19.1 MB


## Progresso dos 6 Dias

| Dia | Tema | Modelo/Ferramenta | Status |
|-----|------|-------------------|--------|
| 1 | Pandas Introdutório | Series & DataFrames | ✅ |
| 2 | Visualização de Dados | Matplotlib & Seaborn | ✅ |
| 3 | Limpeza e Preparação | Tratamento de outliers | ✅ |
| 4 | Regressão Linear | Previsão de preço | ✅ R²=0.92 |
| 5 | Classificação | Random Forest (77% acurácia) | ✅ |
| 6 | Dashboard Interativo | Streamlit app | ✅ |

**Total:** 7 notebooks | 21 docs | 19.1 MB | 53.917 diamantes analisados

In [9]:
import subprocess

# Pega o histórico de commits
resultado = subprocess.run(
    ['git', 'log', '--oneline', '-20'],
    cwd=str(projeto_path),
    capture_output=True,
    text=True
)

print("=== Últimos 20 Commits ===\n")
print(resultado.stdout)

# Conta total de commits
resultado_total = subprocess.run(
    ['git', 'rev-list', '--count', 'HEAD'],
    cwd=str(projeto_path),
    capture_output=True,
    text=True
)

total_commits = resultado_total.stdout.strip()
print(f"Total de commits: {total_commits}")

=== Últimos 20 Commits ===


Total de commits: 


In [10]:
import subprocess
from pathlib import Path

# Caminho absoluto do projeto
projeto_path = Path.home() / "Documentos/projetos/hstern-datascience"

# Pega o histórico de commits
resultado = subprocess.run(
    ['git', 'log', '--oneline', '-20'],
    cwd=str(projeto_path),
    capture_output=True,
    text=True
)

print("=== Últimos 20 Commits ===\n")
print(resultado.stdout)

# Conta total de commits
resultado_total = subprocess.run(
    ['git', 'rev-list', '--count', 'HEAD'],
    cwd=str(projeto_path),
    capture_output=True,
    text=True
)

total_commits = resultado_total.stdout.strip()
print(f"Total de commits: {total_commits}")

=== Últimos 20 Commits ===

eaeb607 chore: adiciona pasta context com documentação de sessão
e22a682 docs: adiciona diário de aprendizado do dia 6
9b2b52f Dia 6: dashboard interativo com Streamlit
54a7331 docs: adiciona diário de aprendizado do dia 4
5050376 Dia 5: classificação com Random Forest — previsão do corte do diamante
06964c8 Dia 4: primeiro modelo de Machine Learning — Regressão Linear
6e2d9b8 Dia 3: limpeza e preparação do dataset de diamantes
932c912 Dia 3: limpeza e preparação do dataset de diamantes
4706ded Dia 2: visualização de dados com matplotlib e seaborn - dataset diamonds
d841074 Adiciona README com resumo completo do dia 1
e3980a9 Dia 1: introducao ao Pandas com dataset de joalheria

Total de commits: 11


## Portfolio no GitHub

**Repositório:** https://github.com/AgapitoFernando/hstern-datascience

### O que impressiona um recrutador da HStern

✅ **Progressão clara** — Dia 1 a Dia 6, cada passo documentado
✅ **Código funcional** — Modelos reais, não exercícios teóricos
✅ **Dados reais** — 53.917 diamantes, limpeza profissional
✅ **Documentação** — 21 arquivos markdown explicando cada conceito
✅ **Commits descritivos** — Histórico claro de decisões
✅ **Diversidade técnica** — Pandas, sklearn, Streamlit, Docker
✅ **Domain knowledge** — Insights sobre joalheria (quilates, cortes)

### Próximas fases (Dias 8-14)

- Dia 8: O que é Gen AI
- Dia 9: APIs de IA na prática
- Dia 10: Prompt Engineering
- Dia 11: Automação com IA
- Dia 12: NLP básico
- Dia 13: Projeto integrador (FAJ + HStern dados)
- Dia 14: Preparação para entrevista

In [11]:
print("=" * 60)
print("DIA 7 — REVISÃO E GITHUB — SUMÁRIO")
print("=" * 60)

print("\n📊 ESTATÍSTICAS DO PROJETO")
print(f"  • Notebooks:        {notebooks}")
print(f"  • Scripts Python:   {scripts_python}")
print(f"  • Documentação:     {markdown_docs} arquivos")
print(f"  • Datasets:         {csv_files} arquivos CSV")
print(f"  • Tamanho total:    {size_mb:.1f} MB")

print("\n📈 PROGRESSO NOS ESTUDOS")
print("  • Dias completados: 6")
print("  • Commits no GitHub: 11")
print("  • Modelos treinados: 2 (Regressão Linear, Random Forest)")
print("  • Dashboard funcional: Sim (Streamlit)")

print("\n🎯 PORTFOLIO")
print("  • Repositório: https://github.com/AgapitoFernando/hstern-datascience")
print("  • Visibilidade: Público no GitHub")
print("  • Pronto para HStern: Sim — demonstra rigor técnico e aprendizado contínuo")

print("\n🚀 PRÓXIMOS PASSOS")
print("  • Dia 8: O que é Gen AI")
print("  • Integração com FAJ e Guiparch")
print("  • Deploy do dashboard Streamlit")

print("\n" + "=" * 60)

DIA 7 — REVISÃO E GITHUB — SUMÁRIO

📊 ESTATÍSTICAS DO PROJETO
  • Notebooks:        7
  • Scripts Python:   8
  • Documentação:     21 arquivos
  • Datasets:         9 arquivos CSV
  • Tamanho total:    19.1 MB

📈 PROGRESSO NOS ESTUDOS
  • Dias completados: 6
  • Commits no GitHub: 11
  • Modelos treinados: 2 (Regressão Linear, Random Forest)
  • Dashboard funcional: Sim (Streamlit)

🎯 PORTFOLIO
  • Repositório: https://github.com/AgapitoFernando/hstern-datascience
  • Visibilidade: Público no GitHub
  • Pronto para HStern: Sim — demonstra rigor técnico e aprendizado contínuo

🚀 PRÓXIMOS PASSOS
  • Dia 8: O que é Gen AI
  • Integração com FAJ e Guiparch
  • Deploy do dashboard Streamlit

